smartsim/settingshold/translators/batch/pbs.py

```diff
@@ -0,0 +1,191 @@
+        t.cast(t.List[str],batch_arg_copy)
+        for opt, value in batch_arg_copy.items():
+            prefix = "-"
+            if not value:
+                raise ValueError("PBS options without values are not allowed")
+            opts += [" ".join((prefix + opt, str(value)))]
+        return opts
+
+    def _sanity_check_resources(
+        self, batch_args: t.Dict[str, t.Union[str,int,float,None]]
+    ) -> None:
+        """Check that only select or nodes was specified in resources
+
+        Note: For PBS Pro, nodes is equivalent to 'select' and 'place' so
+        they are not quite synonyms. Here we assume that
+        """
+        checked_resources = batch_args
+
+        has_select = checked_resources.get("select", None)
+        has_nodes = checked_resources.get("nodes", None)
+
+        if has_select and has_nodes:
+            raise SSConfigError(
+                "'select' and 'nodes' cannot both be specified. This can happen "
+                "if nodes were specified using the 'set_nodes' method and "
+                "'select' was set using 'set_resource'. Please only specify one."
+            )
+
+        if has_select and not isinstance(has_select, int):
+            raise TypeError("The value for 'select' must be an integer")
+        if has_nodes and not isinstance(has_nodes, int):
+            raise TypeError("The value for 'nodes' must be an integer")
+
+        for key, value in checked_resources.items():
+            if not isinstance(key, str):
+                raise TypeError(
+                    f"The type of {key} is {type(key)}. Only int and str "
+                    "are allowed."
+                )
+            if not isinstance(value, (str, int)):
+                raise TypeError(
+                    f"The value associated with {key} is {type(value)}. Only int "
+                    "and str are allowed."
+                )
+
+    def _create_resource_list(self, batch_args: t.Dict[str, t.Union[str,int,float,None]]) -> t.Tuple[t.List[str],t.Dict[str, t.Union[str,int,float,None]]]:
+        self._sanity_check_resources(batch_args)
+        res = []
+
+        batch_arg_copy = deepcopy(batch_args)
+        # Construct the basic select/nodes statement
+        if select := batch_arg_copy.pop("select", None):
+            select_command = f"-l select={select}"
+        elif nodes := batch_arg_copy.pop("nodes", None):
+            select_command = f"-l nodes={nodes}"
+        else:
+            raise SSConfigError(
+                "Insufficient resource specification: no nodes or select statement"
+            )
+        if ncpus := batch_arg_copy.pop("ppn", None):
+            select_command += f":ncpus={ncpus}"
+        if hosts := batch_arg_copy.pop("hostname", None):
+            hosts_list = ["=".join(str(hosts))]
+            t.cast(str,hosts_list)
+            select_command += f":{'+'.join(hosts_list)}"
+        res += [select_command]
```

Comment:
Another request here to isolate each symbol in the command to its own string
```suggestion
        res += select_command.split()
```